sql-files/main.sql

```diff
@@ -1032,3 +1033,50 @@ CREATE TABLE IF NOT EXISTS `npc_barter_data` (
   `priceAmount` INT UNSIGNED NOT NULL DEFAULT '0',
   PRIMARY KEY (`name`, `itemid`, `priceId`, `priceAmount`)
 ) ENGINE=MyISAM;
+
+CREATE TABLE IF NOT EXISTS `npc_expanded_barter_data` (
+  `name` VARCHAR(24) NOT NULL DEFAULT '',
+  `itemId` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `amount` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `zeny` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyId1` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyAmount1` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyRefine1` INT(11) NOT NULL DEFAULT '0',
+  `currencyId2` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyAmount2` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyRefine2` INT(11) NOT NULL DEFAULT '0',
+  `currencyId3` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyAmount3` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyRefine3` INT(11) NOT NULL DEFAULT '0',
+  `currencyId4` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyAmount4` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyRefine4` INT(11) NOT NULL DEFAULT '0',
+  `currencyId5` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyAmount5` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyRefine5` INT(11) NOT NULL DEFAULT '0',
+  `currencyId6` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyAmount6` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyRefine6` INT(11) NOT NULL DEFAULT '0',
+  `currencyId7` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyAmount7` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyRefine7` INT(11) NOT NULL DEFAULT '0',
+  `currencyId8` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyAmount8` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyRefine8` INT(11) NOT NULL DEFAULT '0',
+  `currencyId9` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyAmount9` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyRefine9` INT(11) NOT NULL DEFAULT '0',
+  `currencyId10` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyAmount10` INT(11) UNSIGNED NOT NULL DEFAULT '0',
+  `currencyRefine10` INT(11) NOT NULL DEFAULT '0',
+  PRIMARY KEY (`name`, `itemid`, `zeny`,
+    `currencyId1`, `currencyAmount1`, `currencyRefine1`,
+    `currencyId2`, `currencyAmount2`, `currencyRefine2`,
+    `currencyId3`, `currencyAmount3`, `currencyRefine3`,
+    `currencyId4`, `currencyAmount4`, `currencyRefine4`,
+    `currencyId5`, `currencyAmount5`, `currencyRefine5`,
+    `currencyId6`, `currencyAmount6`, `currencyRefine6`,
+    `currencyId7`, `currencyAmount7`, `currencyRefine7`,
+    `currencyId8`, `currencyAmount8`, `currencyRefine8`
```

Comment:
```suggestion
    `currencyId9`, `currencyAmount9`, `currencyRefine9`,
    `currencyId10`, `currencyAmount10`, `currencyRefine10`
)
```
missing currency 9 and 10?  but does currency, amount, refine need to be primary key? since they could be in random order too?

Reply:
keys already too much. i removed more keys because mysql not accept this amount

Reply:
keys need for simple update queries, it works same as in other shops

Reply:
oic, saw the latest commit, it become 4 keys..